mmv1/templates/terraform/examples/redis_cluster_ha.tf.erb

```diff
@@ -11,6 +11,9 @@ resource "google_redis_cluster" "<%= ctx[:primary_resource_id] %>" {
   depends_on = [
     google_network_connectivity_service_connection_policy.default
   ]
+  lifecycle {
```

Comment:
These configurations are used for a few purposes in addition to docs, so we need this to change based on the environment (test and oics are false, docs is true). See https://github.com/GoogleCloudPlatform/magic-modules/blob/2b59ef214075b85da7c08bbef9a5e02760f5d412/mmv1/products/bigqueryconnection/Connection.yaml#L57-L73 for an example that does this already.

Reply:
deletion_protection appears to be something different than the prevent_destroy metadata configuration though. Are you suggesting that these vars can be used to inject into templates for the tests so that prevent_destroy is not set on the tests also?

Reply:
> Are you suggesting that these vars can be used to inject into templates for the tests so that prevent_destroy is not set on the tests also?

Oh, yes- the relevant bit here is that we want the boolean `true`/`false` to be different between targets, and not about the specific field or lifecycle rule in use. It just so happens that the samples that need to manage this kind of stuff are doing it for similar reasons.

Reply:
I've used the overrides to conditionally render the lifecycle metadata configuration if prevent_destroy is set to 'true'. I noticed that all the values in the Hash objects for vars were strings so I used the strings 'true' and 'false' to keep things consistent.